core/schema.rs

```diff
@@ -576,28 +576,37 @@ fn create_table(
 
                 let mut typename_exactly_integer = false;
                 let ty = match col_def.col_type {
-                    Some(data_type) => {
+                    Some(data_type) => 'ty: {
                         // https://www.sqlite.org/datatype3.html
                         let mut type_name = data_type.name;
                         type_name.make_ascii_uppercase();
-                        if type_name.contains("INT") {
-                            typename_exactly_integer = type_name == "INTEGER";
-                            Type::Integer
-                        } else if type_name.contains("CHAR")
-                            || type_name.contains("CLOB")
-                            || type_name.contains("TEXT")
-                        {
-                            Type::Text
-                        } else if type_name.contains("BLOB") || type_name.is_empty() {
-                            Type::Blob
-                        } else if type_name.contains("REAL")
-                            || type_name.contains("FLOA")
-                            || type_name.contains("DOUB")
-                        {
-                            Type::Real
-                        } else {
-                            Type::Numeric
+
+                        if type_name.is_empty() {
+                            break 'ty Type::Blob;
                         }
+
+                        if type_name == "INTEGER" {
+                            typename_exactly_integer = true;
+                            break 'ty Type::Blob;
+                        }
+
+                        if let Some(ty) = type_name.as_bytes().windows(4).find_map(|s| match s {
```

Comment:
the ordering is of these pattern matches is important https://www.sqlite.org/datatype3.html

the "contains INT" check should come first, etc.

```
If the declared type contains the string "INT" then it is assigned INTEGER affinity.

If the declared type of the column contains any of the strings "CHAR", "CLOB", or "TEXT" then that column has TEXT affinity. Notice that the type VARCHAR contains the string "CHAR" and is thus assigned TEXT affinity.

If the declared type for a column contains the string "BLOB" or if no type is specified then the column has affinity BLOB.

If the declared type for a column contains any of the strings "REAL", "FLOA", or "DOUB" then the column has REAL affinity.

Otherwise, the affinity is NUMERIC.
```